run_nodes.sh

```diff
@@ -84,11 +89,14 @@ for i in $(seq 0 "$(( N_VALIDATORS + N_NON_VALIDATORS - 1 ))"); do
     --ws-port $((9944 + i)) \
     --port $((30334 + i)) \
     --bootnodes $bootnodes \
+    --public-addr ${addresses[i]} \
```

Comment:
This looks like a red flag to me that something went wrong... Why is this necessary? And what it does internally?

Reply:
When running purely locally the substrate network does not return any own addresses ever. This flag makes it at least know one such address.

We expect everything should work without this flag in any non-local context, but we will have to test this extensively ofc.

Reply:
@timorl is right, can this stay or we should change something?

Reply:
If you run a node locally, even with this flag it normally logs something like:
`
2022-01-13 12:04:35.149  INFO                 main sub-libp2p: 🔍 Discovered new external address for our node: /dns4/localhost/tcp/30335/p2p/12D3KooWBhABJ8f8LcoJPMyFtJTZnE6umqgHGQmcMhVLHSjKzoVS/p2p/12D3KooWBhABJ8f8LcoJPMyFtJTZnE6umqgHGQmcMhVLHSjKzoVS 
`
wouldn't this be exactly what we need?

If I remove this flag, then it never works, or just kicks in later?

Reply:
Hmm, you are right, this appears, in my case `2022-01-14 16:50:30.513  INFO tokio-runtime-worker sub-libp2p: 🔍 Discovered new external address for our node: /ip4/127.0.0.1/tcp/30334/p2p/12D3KooWAYdQQNDwFw1QSWcbUSLFz2Tk6t5UWkVh2FiD8RPetYFK`. Then maybe this doesn't work after all and we have to figure out why the substrate network does not want to tell us about this address. :/

Reply:
Welp, found it, substrate network does find this address with the `p2p` part, but afterwards returns it without. It also returns the peer id separately, so there is a somewhat aggressive fix possible, maybe I'll make a PR deeper in the evening. <_<"

Reply:
This flag can (and probably should) be removed now.